esp/services/ws_loggingservice/loggingservice.cpp

```diff
@@ -25,39 +25,76 @@
 
 typedef IEspLogAgent* (*newLogAgent_t_)();
 
-bool CWsLoggingServiceEx::init(const char* service, const char* type, IPropertyTree* cfg, const char* process)
+void CWsLoggingServiceEx::initLogAgent(IPropertyTree &ptree, const char *process, const char *service)
 {
-    VStringBuffer xpath("Software/EspProcess[@name=\"%s\"]/EspService[@name=\"%s\"]", process, service);
-    Owned<IPropertyTree> pServiceNode = cfg->getPropTree(xpath.str());
-    if (!pServiceNode)
-        throw MakeStringException(-1, "No settings found for service %s", service);
+    const char* agentName = ptree.queryProp("@name");
+    const char* agentType = ptree.queryProp("@type");
+    const char* agentPlugin = ptree.queryProp("@plugin");
+    if (!agentName || !*agentName || !agentPlugin || !*agentPlugin)
+        return;
+
+    IEspLogAgent* logAgent = loadLoggingAgent(agentName, agentPlugin);
+    if (!logAgent)
+    {
+        OERRLOG(-1, "Failed to create logging agent for %s", agentName);
+        return;
+    }
+    logAgent->init(agentName, agentType, &ptree, process);
```

Comment:
Something missing in the original version, which is my fault because I didn't recognize this service as a manager of log agents, is a call to the agent's initVariants method. There is not a common init method shared by agents, so my addition of variants to agents yielded a a separate, shared method.

If we could add `logAgent->initVariants(&ptree);` immediately after `init` we will improve agent compatibility.

Reply:
I'll add it.